src/coreclr/vm/ceeload.h

```diff
@@ -914,6 +916,31 @@ class Module : public ModuleBase
         _ASSERTE((m_dwTransientFlags & IS_EDIT_AND_CONTINUE) == 0 || IsEditAndContinueCapable());
         return (m_dwTransientFlags & IS_EDIT_AND_CONTINUE) != 0;
     }
+    
+    BOOL IsInliningDisabledByProfiler() const {WRAPPER_NO_CONTRACT; SUPPORTS_DAC; return (m_dwTransientFlags & PROF_DISABLE_INLINING) != 0; }
+
+    BOOL AreJITOptimizationsDisabled() const
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        #ifdef DEBUGGING_SUPPORTED
+        // check if debugger has disallowed JIT optimizations
+        if(!CORDebuggerAllowJITOpts(GetDebuggerInfoBits()))
+        {
+            return TRUE;
+        }
+        #endif
+
+        #if defined(PROFILING_SUPPORTED) || defined(PROFILING_SUPPORTED_DATA)
+        // check if profiler had disabled JIT optimizations when module was loaded
+        if(m_dwTransientFlags & PROF_DISABLE_OPTIMIZATIONS){
+            return TRUE;
+        }
+        #endif
```

Comment:
To keep the style consistent, untab preprocessor directives + spacing after if + `{` next line.
```suggestion
#ifdef DEBUGGING_SUPPORTED
        // check if debugger has disallowed JIT optimizations
        if (!CORDebuggerAllowJITOpts(GetDebuggerInfoBits()))
        {
            return TRUE;
        }
#endif

#if defined(PROFILING_SUPPORTED) || defined(PROFILING_SUPPORTED_DATA)
        // check if profiler had disabled JIT optimizations when module was loaded
        if (m_dwTransientFlags & PROF_DISABLE_OPTIMIZATIONS)
        {
            return TRUE;
        }
#endif
```

Reply:
If the check is under `defined(PROFILING_SUPPORTED) || defined(PROFILING_SUPPORTED_DATA)`, then should bit be set under the same condition? in ceeload.cpp its only `#ifdef PROFILING_SUPPORTED`

Reply:
> If the check is under defined(PROFILING_SUPPORTED) || defined(PROFILING_SUPPORTED_DATA), then should bit be set under the same condition? in ceeload.cpp its only #ifdef PROFILING_SUPPORTED

Using PROFILING_SUPPORTED only for the write and PROFILING_SUPPORTED || PROFILING_SUPPORTED_DATA for the read seems reasonable to me. PROFILING_SUPPORTED_DATA is used in the DAC where reading existing data structures is expected but initializing/writing them is not. The code would work correctly if PROFILING_SUPPORTED_DATA is added in the initialization path but it might create some confusion because it implies the init code might be executed by the DAC.

Reply:
I actually used the same condition as what was controlling the original macro. I am not familiar enough with the use of precompilation symbols in the runtime to have a strong opinion about it. If you think a change needs to be made, I'll be happy to do it.

Reply:
I don't think a change is needed :)